lib/yang.h

```diff
@@ -316,6 +336,21 @@ extern void yang_dnode_get_path(const struct lyd_node *dnode, char *xpath,
 extern const char *yang_dnode_get_schema_name(const struct lyd_node *dnode,
 					      const char *xpath_fmt, ...);
 
+/*
+ * Find a libyang data node by its YANG data path.
+ *
+ * dnode
+ *    Base libyang data node to operate on.
+ *
+ * xpath
+ *    XPath expression (absolute or relative).
+ *
+ * Returns:
+ *    The libyang data node if found, or NULL if not found.
+ */
+extern struct lyd_node *yang_dnode_pget(const struct lyd_node *dnode,
+					const char *xpath);
```

Comment:
Correct, and yes. This is a non-variadic function, the places that call this have already done a vsnprintf on variadic arguments, to do so again is computationally wasteful, and technically could be wrong. Consider a format string `"%%stop%%"`. Once through vsnprintf will turn this into `"%stop%"`, send that through another vsnprintf and it will try and read a string from varargs and may crash/corrupt memory.